test/variational/test_nearest_neighbor_variational_strategy.py

```diff
@@ -115,7 +115,7 @@ def _training_iter(
         return output, loss
 
     def _eval_iter(self, model, cuda=False):
-        inducing_batch_shape = model.variational_strategy.inducing_points.shape[:-2]
+        inducing_batch_shape = model.variational_strategy._inducing_batch_shape
```

Comment:
Can you add a unit test that runs VNNGP with batches? AFIAK, this unit test is still only running for non-batched VNNGP.

Reply:
I think line 194 `test_training_iteration_batch_model` does the batch model testing.

Reply:
@LuhuanWu why wasn't `test_training_iteration_batch_model` failing before this PR then?
Ideally we want to have a test case that (1) would've failed before this PR was added (capturing the behavior described in #2300) but (2) does not fail with the new code in this PR.

Reply:
good point. I'll look into that.